### PR TITLE
added 2020-01-13-preview

### PR DESCRIPTION
**Pull request by juliaweingart**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ArcturusZhang** [comment]
Hi @juliaweingart could you please fix the CI errors before we could review this?
And also the SDK generation fails in every language because the modeler reports some errors for this swagger, you may also need to resolve this. Thanks

**Comment by juliaweingart** [comment]
Hi Arcturus,

Thanks for reaching out. I have resolved all errors other than the SDK automation errors as I am not sure how to resolve these. Based on the logs in the pipeline<https://dev.azure.com/azure-sdk/internal/_build/results?buildId=541865&view=logs&j=b6885ff5-5b63-57e7-07bc-9d8d90d734f1&t=edefe71f-a8fe-52c2-c45d-59609a33c1f2>, I see a ‘permission denied’ error at this step: Installing AutoRest extension '@microsoft.azure/autorest.modeler'. Any idea what this could mean?

Thanks,
Julia

**Comment by ArcturusZhang** [comment]
Hi @juliaweingart the pipeline for dot net could get a `permission denied` occasionally (maybe something wrong with the pipeline itself), please check the error messages from other languages, for instance Golang [here](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=541865&view=logs&j=fae805ed-8bde-5aa7-6387-2396f34533f9&t=7678b48e-bcab-501c-34d6-7d8f27ecbef0). You could see an exception was thrown during the modeler. 

An error thrown in the modeler usually means that the swagger does not agree with itself in some of the model definition - but yeah, just based on this error message, I cannot identify exactly which model is causing trouble.

**Comment by ArcturusZhang** [comment]
Also @juliaweingart could you please also modify your commit history by making one previous api-version the first commit?
ARM review usually needs to compare the new api-version with the old api-version, and it helps a lot if you just copy the old api-version to this directory and we could just compare these api-versions by selecting a range of commit. 

And this may also help identify the modeler error.

**Comment by juliaweingart** [comment]
Hi Acturus,

2 questions –

  1.  Are you asking me to squash my commits into one larger commit that is “adding an api version”? The “old api-version” is under this directory -- azure-rest-api-specs<https://github.com/Azure/azure-rest-api-specs/tree/dd122c8a00723488842768c267f68aded51013c6>/specification<https://github.com/Azure/azure-rest-api-specs/tree/dd122c8a00723488842768c267f68aded51013c6/specification>/workloadmonitor<https://github.com/Azure/azure-rest-api-specs/tree/dd122c8a00723488842768c267f68aded51013c6/specification/workloadmonitor>/resource-manager<https://github.com/Azure/azure-rest-api-specs/tree/dd122c8a00723488842768c267f68aded51013c6/specification/workloadmonitor/resource-manager>/Microsoft.WorkloadMonitor<https://github.com/Azure/azure-rest-api-specs/tree/dd122c8a00723488842768c267f68aded51013c6/specification/workloadmonitor/resource-manager/Microsoft.WorkloadMonitor>/preview<https://github.com/Azure/azure-rest-api-specs/tree/dd122c8a00723488842768c267f68aded51013c6/specification/workloadmonitor/resource-manager/Microsoft.WorkloadMonitor/preview>/2018-08-31-preview<https://github.com/Azure/azure-rest-api-specs/tree/dd122c8a00723488842768c267f68aded51013c6/specification/workloadmonitor/resource-manager/Microsoft.WorkloadMonitor/preview/2018-08-31-preview>/Microsoft.WorkloadMonitor.json
  2.  With regards to the SDK error, when looking deeper into the logs, I see the following error messages:
     *   Failure during batch task - {"tag":"package-2020-01-13-preview"}, FATAL: System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.'
     *   This specifically seems to be referring to the readme<https://github.com/Azure/azure-rest-api-specs/pull/10817/files#diff-c29721e7d47ff59ad59f5e10a9ce2bd7> describing SDK generation settings on lines 140 and 180, where I updated the api version to the new one. Any idea what is going on here?

Thanks,
Julia

**Comment by ArcturusZhang** [comment]
Hi @juliaweingart since we need to compare this new api-version with the old api-version, and github does not provide a functionality that you could compare between directories, therefore we have to comparing these different api-versions via commits. By making the copy of the old api-version the first commit in this PR, we could easily get the differences between these two api-versions by selecting all other commits except the first commit, so that we do not have to review the whole api-version as a new.

File: specification/workloadmonitor/resource-manager/readme.md [hunk 1]
```diff
@@ -122,7 +137,7 @@ go:
 
 ``` yaml $(go) && $(multiapi)
 batch:
-  - tag: package-2018-08-31-preview
+  - tag: package-2020-01-13-preview
```
Comment (ArcturusZhang) [inline specification/workloadmonitor/resource-manager/readme.md:140]: Please add the new tag to this list and do not remove the old tag.
```suggestion
  - tag: package-2018-08-31-preview
  - tag: package-2020-01-13-preview
```

File: specification/workloadmonitor/resource-manager/readme.md [hunk 2]
```diff
@@ -153,7 +177,7 @@ output-folder: $(azure-libraries-for-java-folder)/azure-mgmt-workloadmonitor
 
 ``` yaml $(java) && $(multiapi)
 batch:
-  - tag: package-2018-08-31-preview
+  - tag: package-2020-01-13-preview
```
Comment (ArcturusZhang) [inline specification/workloadmonitor/resource-manager/readme.md:180]: Please add the new tag to this list and do not remove the old tag.
```suggestion
  - tag: package-2018-08-31-preview
  - tag: package-2020-01-13-preview
```

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]
About the autorest errors, I do not know why this happens, but I left some comments on the multiapi parts in the readme, maybe resolving this could solve this issue. Please have a look, thanks

**Comment by juliaweingart** [comment]
Okay, I am happy to make this commit. However, the two api versions are completely different, and I’m not sure there would be any use in comparing them. We completely rewrote our resource provider and redesigned the API. Rather than define a new resource type and create a new swagger/manifest/etc, we decided to differentiate the two RP services via their API version. We will be deprecating the old RP service along with the older API version in a few months when we launch public preview.

The more useful comparison, however, would be with my PR into your private repo<https://github.com/Azure/azure-rest-api-specs-pr/pull/1153> of this same new API version that received ARM sign off a few months ago. There have only been a few updates/changes since then that should need review. It might make more sense to start with a copy paste of this signed-off version as the first commit, and then you would be able to more easily review the additional changes that I have added in the past few months?

Regarding the SDK errors – I made the changes you suggested but am still seeing the same error messages… Any other ideas or advice on how to debug this?

**Review by pilor — APPROVED** [review]

**Comment by juliaweingart** [comment]
As an update here, I touched base with Chris Eggert, who reviewed our PR into the private repo, and informed him of the updates I made for this PR. He has reviewed and approved the changes. Now, I believe the last step here is to resolve these SDK automation errors.

From: Julia Weingart
Sent: Thursday, September 24, 2020 11:53 AM
To: Azure/azure-rest-api-specs <reply@reply.github.com>; Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>
Cc: Mention <mention@noreply.github.com>
Subject: RE: [Azure/azure-rest-api-specs] added 2020-01-13-preview (#10817)

Okay, I am happy to make this commit. However, the two api versions are completely different, and I’m not sure there would be any use in comparing them. We completely rewrote our resource provider and redesigned the API. Rather than define a new resource type and create a new swagger/manifest/etc, we decided to differentiate the two RP services via their API version. We will be deprecating the old RP service along with the older API version in a few months when we launch public preview.

The more useful comparison, however, would be with my PR into your private repo<https://github.com/Azure/azure-rest-api-specs-pr/pull/1153> of this same new API version that received ARM sign off a few months ago. There have only been a few updates/changes since then that should need review. It might make more sense to start with a copy paste of this signed-off version as the first commit, and then you would be able to more easily review the additional changes that I have added in the past few months?

Regarding the SDK errors – I made the changes you suggested but am still seeing the same error messages… Any other ideas or advice on how to debug this?

**Comment by ArcturusZhang** [comment]
Hi @juliaweingart if this new api-version is totally different with any previous api-versions, we do not need to alter the commit history to achieve a comparison with the old api-version then.

I am also confused on the SDK failures. Let me ask for someone from modeler to help.

**Review by markcowl — APPROVED** [review]

**Comment by markcowl** [comment]
Closing in favor of #11039

**Pull request closed without merging**